### ISSUE TITLE
Would it be possible to code not used in the SDK into a separate package?

### ISSUE THREAD

**Issue by realityking** [netlify/zip-it-and-ship-it#28]
**- Do you want to request a *feature* or report a *bug*?**

Feature request, kinda.

**- What is the current behavior?**

We depend on Netlify's `js-client` in multiple repositories for some advanced deployment voodoo. Currently this pulls in a lot of dependencies, notably `express` (quite big) and `chokidar` (a native dependency) but also a cli (`minimist`). None of this functionality seems to be used by `js-client` and it's rather surprising that an SDK comes with such large dependencies.

**- If the current behavior is a bug, please provide the steps to reproduce.**

**- What is the expected behavior?**

It'd be great to separate out the code that is required by `js-client` (or vice versa) to keep the SDK small and light.

**Comment by bcomnes** [comment]
Maybe.  

- serve.js could thematically be moved to the dev plugin.  This is what carries the larger deps like chokidar and express.  The dev plugin is already fairly dependency heavy, so that cost has already been paid for in some respects.  @sw-yx what do you think?
- minimist and cliclopts are dep free themsevles, and are small to begin with.   We need to use the dependency resolution algorithm in ZISI in a few contexts and ship a minimal CLI in order to do that.  It's a few bytes. https://github.com/netlify/zip-it-and-ship-it/blob/master/bin.js It could be broken out, too, but at the time it seemed acceptable to keep them together.
- js-client was written with the CLI in mind, so primarily as a Node.js dependency. The browser compatibility was a bonus at the time.  The largest part of the js-client is the https://github.com/netlify/js-client#promisedeploy--clientdeploysiteid-builddir-opts deploy method which has to do a lot of work (and where ZISI is required).  

tldr: 

- The serve concerns are consumed by the dev plugin.  I think it would be fair to move that over there, along with the heavy requirements.  
- The CLI in ZISI is super duper small and shouldn't be cause of heaviness you are feeling.

Thoughts?

**Comment by realityking** [comment]
Makes sense 👍

The CLI is a really minor thing, chokidar and express are what’s i actually care about 🙂

**Comment by sw-yx** [comment]
i like this in principle too. just gotta actually try it :)

**Comment by sw-yx** [comment]
@realityking can you try https://github.com/netlify/js-client/releases/tag/v2.4.2 ?

**Comment by realityking** [comment]
Amazingly fast turn around. Thank you!

@sw-yx Unfortunately that release didn't work out. You reverted, I assume accidentally, the ZISI bump with this commit: https://github.com/netlify/js-client/commit/65cf466a2a66b14a6b63f56cf56ad2b26c46b10c#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

**Comment by sw-yx** [comment]
goddamn merge conflicts... 1 sec

**Comment by sw-yx** [comment]
someday we'll get it right... https://github.com/netlify/js-client/releases/tag/v2.4.3 @realityking

**Comment by realityking** [comment]
That did the trick. Thank you @sw-yx!
